qlib/rl/order_execution/strategy.py

```diff
@@ -146,3 +156,177 @@ def reset(self, outer_trade_decision: BaseTradeDecision = None, **kwargs: Any) -
             order_list = outer_trade_decision.order_list
             assert len(order_list) == 1
             self._order = order_list[0]
+
+
+class SAOEIntStrategy(SAOEStrategy):
```

Comment:
What's the difference between this and SAOEStrategy? Why do we need another class?

Reply:
`SAOEStrategy` is the strategy that maintains `SAOEState` internally (by `SAOEStateAdapter`). `SAOEIntStrategy` is a subclass of `SAOEStrategy`, but it contains interpreters. For now, `SAOEIntStrategy` is the only implementation of `SAOEStrategy` but I think it is useful to leave this two-level structure for future scalability.

Reply:
Better to explain that in docstring.